### PR TITLE
Move TLS to nethserver-mail-common

### PR DESCRIPTION
**Pull request by stephdl**
Use case is: Google MTA does a SMTP transaction with NethSecurity SMTP proxy. No TLS is offered ATM.

We need to move all smtpd_tls* to nethserver-mail-common

https://github.com/NethServer/dev/issues/6219

### CODE REVIEW

File: common/etc/e-smith/templates/etc/postfix/main.cf/30smtpdTLS [hunk 1]
```diff
@@ -1,20 +1,14 @@
...
-smtpd_sasl_tls_security_options = $smtpd_sasl_security_options
 smtpd_tls_auth_only = yes
-smtpd_sasl_authenticated_header = yes
 
 # Use default dovecot cert+key files:
 smtpd_tls_cert_file = /etc/pki/dovecot/certs/dovecot.pem
 smtpd_tls_key_file = /etc/pki/dovecot/private/dovecot.pem
 
-# Use specific DH group 2048 bit, renewed each week by cron
+# Use specific DH group 2048 bit, to renew it, remove the key and restart postfix
```
Comment (DavidePrincipi) [inline common/etc/e-smith/templates/etc/postfix/main.cf/30smtpdTLS:11]:
```suggestion
# Use specific DH group 2048 bit, to renew it, remove the file and restart postfix
```

Comment (stephdl) [reply common/etc/e-smith/templates/etc/postfix/main.cf/30smtpdTLS]: we use now the postfix key for TLS

File: common/usr/lib/systemd/system/postfix-create-DH2048.service [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [Unit]
 Description=Create the DH2048 group for postfix
-ConditionPathExists=!/etc/pki/dovecot/private/postfix.dh.param
+ConditionPathExists=!/etc/postfix/postfix.dh.param
```
Comment (DavidePrincipi) [inline common/usr/lib/systemd/system/postfix-create-DH2048.service:4]: If we change the path we need to generate a new group.

What about moving the existing one (if we have it) to the new location?

Comment (stephdl) [reply common/usr/lib/systemd/system/postfix-create-DH2048.service:4]: DH is ephemeral, at least we could remove it....but does it worth it

Comment (stephdl) [reply common/usr/lib/systemd/system/postfix-create-DH2048.service:4]: We clean the old DH group

Comment (stephdl) [reply common/usr/lib/systemd/system/postfix-create-DH2048.service:4]: hello @DavidePrincipi, could you check now the old group is removed if find

**Pull request merged**